File: bsp4rs/src/com/jetbrains/bsp/bsp4rs/bsp/initialize_build_params.rs
use serde::{Deserialize, Serialize};

use crate::*;

#[derive(Debug, PartialEq, Clone, Default, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct InitializeBuildParams {
    /** Name of the client */
    #[serde(default)]
    pub display_name: String,
    /** The version of the client */
    #[serde(default)]
    pub version: String,
    /** The BSP version that the client speaks */
    #[serde(default)]
    pub bsp_version: String,
    /** The rootUri of the workspace */
    #[serde(default)]
    pub root_uri: URI,
    /** The capabilities of the client */
    #[serde(default)]
    pub capabilities: BuildClientCapabilities,
    /** Additional metadata about the client */
    #[serde(default, flatten, skip_serializing_if = "Option::is_none")]
    pub data: Option<InitializeBuildParamsData>,
}

File: bsp4kt/src/org/jetbrains/bsp/bsp4kt/InitializeBuildParams.kt
package org.jetbrains.bsp.bsp4kt

import kotlinx.serialization.Serializable
import kotlinx.serialization.SerialName
import kotlinx.serialization.json.JsonElement

@Serializable
data class InitializeBuildParams(
  val displayName: String,
  val version: String,
  val bspVersion: String,
  val rootUri: URI,
  val capabilities: BuildClientCapabilities,
  val dataKind: String? = null,
  val data: JsonElement? = null
)
